### PR TITLE
Fix an issue that celery inpect api can not get complete args via the…

### PR DESCRIPTION
**Pull request by rhuab**
#5513

### CODE REVIEW

**Comment by rhuab** [comment]
In celery.worker.control, the function "active(state, **kwargs)" is used to generate a reply for command "active". It calls "requests.info()" where the bug exists

**Comment by rhuab** [comment]
@auvipy Could you tell me why my PR get py2.7 error on CI job? I am not very familiar with py2.7 and do not know the test rule in details. Sorry for bothering, but I need some help.

**Comment by rhuab** [comment]
> you will need to add failing test to get a proper review.

Sorry I don't understand what you exactly mean. It seems that py2.7 test failed at a place not related to my modification and a Travis job failed at downloading py3.5

**Comment by auvipy** [comment]
that's a different case, but to get your PR accepted you will need to add a proper unit test. so focus on that first, please.

**Comment by rhuab** [comment]
> that's a different case, but to get your PR accepted you will need to add a proper unit test. so focus on that first, please.

I will do that

File: celery/worker/request.py [hunk 1]
```diff
@@ -77,100 +77,240 @@ class Request(object):
...
             'reply_to': properties.get('reply_to'),
             'correlation_id': properties.get('correlation_id'),
-            'delivery_info': {
-                'exchange': delivery_info.get('exchange'),
-                'routing_key': delivery_info.get('routing_key'),
-                'priority': properties.get('priority'),
-                'redelivered': delivery_info.get('redelivered'),
-            }
-
+            'hostname': self._hostname,
```
Comment (thedrow) [inline celery/worker/request.py]: Why have you removed this?
I don't think it's a good idea because Celery expects this elsewhere.

**Comment by rhuab** [comment]
@auvipy 
I add the following features to celery.app.worker.request:
1. Set Request object members as protected (starts with '_'), but for compatibility sake I also set property and setter function for them
2. Update self._request_dict['args'] and self._request_dict['kwargs'] in constructor instead of on the request being executed. The dict is updated by reference rather than full copy to avoid memory usage burst in some situations.
3. Request.info() function returns complete information of task args and kwargs, not in incomplete format.
4. Add some tests to the project

File: celery/worker/request.py (hunk 1 above)
Comment (rhuab) [reply celery/worker/request.py]: Actually I don't remove any member of this class, just add two new members "args" and "kwargs" to maintain complete information of args and kwargs sent by Producer.

**Comment by rhuab** [comment]
Any suggestions?

File: celery/worker/request.py (hunk 1 above)
Comment (thedrow) [reply celery/worker/request.py]: But you removed the `delivery_info` key.

Comment (rhuab) [reply celery/worker/request.py]: self._delivery_info is created at line 155, and is appened to request_dict at line 165. It is exposed as property "delivery_info" at line 173 as well.

**Comment by rhuab** [comment]
Any other question?

**Comment by thedrow** [comment]
None.
I'll review this again.

**Comment by thedrow** [comment]
Huh? I didn't close this issue.

**Comment by thedrow** [comment]
That was a git mishap.
Sorry... I'm fixing it.

**Comment by thedrow** [comment]
Merged in 0f1883374b788302f5bfb8e1fdcf9c13b199b17c.

**Pull request closed without merging**